src/OpenTK.Windowing.GraphicsLibraryFramework/Input/KeyboardState.cs

```diff
@@ -186,7 +186,10 @@ public override string ToString()
 
         internal void Update()
         {
-            _keysPrevious = (BitArray)_keys.Clone();
+            foreach (var index in (int[])Enum.GetValues(typeof(Keys)))
+            {
+                _keysPrevious[index] = _keys[index];
+            }
```

Comment:
I would just copy the entire bit array instead of getting the known Keys values. There may exist keys outside of the enum values.
Another alternative would be to call `BitArray.SetAll(false)` followed by `BitArray.Or(BitArray)` to copy all of the data.

Reply:
Both of those make perfect sense - thanks for the suggestion. I think SetAll(false) followed by Or(_keys) would suit both cases.

Reply:
If we're thinking about keys outside the Enum, what about changing the loop in ToString at the same time?
```cs
foreach (Keys key in (Keys[])Enum.GetValues(typeof(Keys)))
```